src/parser.cpp

```diff
@@ -285,14 +285,17 @@ Writer<std::vector<AST::AST*>> Parser::parse_argument_list() {
  * Here is an article with more information:
  * https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html
  */
-Writer<AST::AST*> Parser::parse_expression(int bp) {
+Writer<AST::AST*> Parser::parse_expression(int bp, AST::AST* parsed_lhs) {
 	Writer<AST::AST*> result = {
 	    {"Parse Error: Failed to parse expression"}};
 
 	Writer<AST::AST*> lhs;
-
-	lhs = parse_terminal();
-	CHECK_AND_RETURN(result, lhs);
+	if (!parsed_lhs) {
+		lhs = parse_terminal();
+		CHECK_AND_RETURN(result, lhs);
+	} else {
+		lhs = make_writer(parsed_lhs);
+	}
```

Comment:
I didn't quite understand why this was needed at first, but I do now. That's some good thinking!